File: src/leetcode/python/easy/RangeSumQueryImmutable_303.py
class RangeSumQueryImmutable_303(object):
    def __init__(self, nums):
        """
        initialize your data structure here.
        :type nums: List[int]
        """
        for i in range(1, len(nums)):
            nums[i] += nums[i-1]
        self.nums = nums

    def sumRange(self, i, j):
        """
        sum of elements nums[i..j], inclusive.
        :type i: int
        :type j: int
        :rtype: int
        """
        if i == 0:
            return self.nums[j]
        return self.nums[j] - self.nums[i-1]

File: src/leetcode/java/RangeSumQueryImmutable_303.java
package leetcode.java.easy;

public class RangeSumQueryImmutable_303 {
  private int[] nums;

  public RangeSumQueryImmutable_303(int[] nums) {
    for (int i = 1; i < nums.length; i++)
      nums[i] += nums[i - 1];
    this.nums = nums;
  }

  public int sumRange(int i, int j) {
    if (i == 0) return nums[j];
    return nums[j] - nums[i - 1];
  }
}
